src/rgw/rgw_sal_motr.cc

```diff
@@ -3877,25 +3918,14 @@ int MotrMultipartUpload::init(const DoutPrefixProvider *dpp, optional_yield y,
     ent.meta.mtime = ceph::real_clock::now();
     ent.meta.user_data.assign(mpbl.c_str(), mpbl.c_str() + mpbl.length());
     ent.encode(bl);
-    std::unique_ptr<RGWObjTags> obj_tags;
     req_state *s = (req_state *) obj_ctx->get_private();
-    /* handle object tagging */
-    // Verify tags exists and add to attrs
-    if (s->info.env->exists("HTTP_X_AMZ_TAGGING")){
-      auto tag_str = s->info.env->get("HTTP_X_AMZ_TAGGING");
-      obj_tags = std::make_unique<RGWObjTags>();
-      int ret = obj_tags->set_from_string(tag_str);
-      if (ret < 0){
-        ldpp_dout(dpp, 0) << "setting obj tags failed with rc=" << ret << dendl;
-        if (ret == -ERR_INVALID_TAG){
-          ret = -EINVAL; //s3 returns only -EINVAL for PUT requests
-        }
-        return ret;
-      }
-      bufferlist tags_bl;
-      obj_tags->encode(tags_bl);
-      attrs[RGW_ATTR_TAGS] = tags_bl;
+    bufferlist tags_bl;
+    int r = parse_tags(dpp, tags_bl, s);
```

Comment:
before calling parse_tags(), please add a debug log saying 'parsing tag values for object : {obj name}'

Reply:
Added